index.html

```diff
@@ -1,49 +1,256 @@
+        {"state":"SD","rate":10.5},
+        {"state":"TN","rate":14.5},
+        {"state":"TX","rate":10.8},
+        {"state":"UT","rate":10.6},
+        {"state":"VA","rate":8.5},
+        {"state":"VT","rate":24.9},
+        {"state":"WA","rate":26.5},
+        {"state":"WI","rate":10.5},
+        {"state":"WV","rate":10.9},
+        {"state":"WY","rate":13.0}
+      ]; 
+
+      var MAX_RATE = 49.3; 
+
+      var options = {
+      	sorted: false, 
+      	filtered: false, 
+      	page: 0
+      }; 
+
+      var tbody = d3.select("tbody"); 
+
+      function updateTable() {
+
+        tbody.html(""); 
+
+        var data = states.slice()
+
+        if (options.filtered === 'filter-r') { 
+            data = data.filter(function (a) { 
+            return a.rate > 15}); 
+        }
+
+        var rows = tbody.selectAll("tr")
+          .data(states)
+          .enter()
+          .append("tr"); 
+
+        rows.append("td")
+          .html(function(d) {return d.state}); 
+        
+        rows.append("td")
+          .html(function (d) {return d.rate}); 
+
+        rows.append("td")
+          .append("div")
+          .classed("bar", true)
+          .style("width", function (d) {return ((d.rate / MAX_RATE) *100) + '%'})
+
+        }; 
+
+        updateTable(); 
+
+        d3.select('#sort-asc')
+          .on("click", function() {
+
+            if (d3.select('#sort-asc').classed('active') === false) {
+              d3.select('#sort-asc').classed('active', true);
+              d3.select('#sort-desc').classed('active', false); 
+              d3.selectAll("tbody tr").sort(function (a, b) {
+                return d3.ascending(a.rate, b.rate); 
+              }); 
+            } else { 
+              d3.select('#sort-asc').classed('active', false); 
+
+            updateTable(); 
+            }
+
+          }); 
+
+        d3.select('#sort-desc')
+          .on("click", function() {
+
+            if (d3.select('#sort-desc').classed('active') === false) {
+              d3.select('#sort-desc').classed('active', true);
+              d3.select('#sort-asc').classed('active', false); 
+              d3.selectAll("tbody tr").sort(function (a, b) {
+                return d3.descending(a.rate, b.rate); 
+              }); 
+            } else { 
+              d3.select('#sort-desc').classed('active', false); 
+
+            updateTable(); 
+            }
+
+          }); 
+
+        d3.select('#filter-byrate')
+          .on("click", function() {
+            if (options.filtered === 'filter-r') {
+                d3.select('#filter-byrate').classed('active', false); 
+                options.filtered = false;
+            }  else {
+               options.filtered = 'filter-r'; 
+               d3.select('#filter-byrate').classed('active', true); 
+               console.log("check"); //nothing happens
+
+            }
+            updateTable(); 
+            
+          }); 
+
+        updateTable(); 
+
```

Comment:
Note you're running updateTable() twice as the page loads - once here and once on line 201. You can lose one of these function calls.